include/openPMD/Dataset.hpp

```diff
@@ -52,4 +52,27 @@ class Dataset
     std::string compression;
     std::string transform;
 };
-} // openPMD
+
+/**
+ * A chunk consists of its offset, its extent
+ * and the rank from which it was written.
+ * If not specified explicitly, the rank will be assumed to be 0.
+ */
+struct Chunk
+{
+    Offset offset;
+    Extent extent;
+    unsigned int rank = 0;
```

Comment:
Can you please document these fundamental members, i.e. do we really need to store the (MPI?) rank here?
I think a fundamental offset/extent Chunk class without those info would be better. Containers around it could then connect it to a rank.

```suggestion
    Offset offset;  //!< origin of the chunk
    Extent extent;  //!< size of the chunk
    unsigned int rank = 0;  //!< ... what is a rank here?
```

General note: best write `mpi_rank` when you mean MPI ranks, because for 2D-arrays, people can mismatch it with [the rank of a matrix](https://en.wikipedia.org/wiki/Rank_(linear_algebra)) in English or with the [HDF5 rank](https://support.hdfgroup.org/HDF5/doc/H5.intro.html) (aka dimension/ndim).

Reply:
The MPI rank is good to know for data distribution in streaming setups, see also [this issue](https://github.com/ornladios/ADIOS2/issues/1679) in ADIOS2 which resulted in the addition of [`Variable<T>::Info::WriterID`](https://github.com/ornladios/ADIOS2/blob/4b24a85077bd43d2eced60f963c9493aa1d97014/bindings/CXX11/adios2/cxx11/Variable.h#L345). In a way, this `Chunk` struct is supposed to be openPMD's variant of `Variable<T>::Info`.

I agree that having a separate, more basic `Chunk` class might be helpful. What about defining a basic `Chunk` class and deriving from it a `WrittenChunk` or something?

Reply:
Sounds great, agreed :)